sdk/@launchdarkly/observability-python/ldobserve/observe.py

```diff
@@ -0,0 +1,261 @@
+import contextlib
+import datetime
+import logging
+import typing
+from opentelemetry.context import Context
+from opentelemetry.instrumentation.logging import LEVELS
+from opentelemetry.sdk._logs import LoggingHandler
+from opentelemetry.trace import Span, Tracer
+import opentelemetry.trace as trace
+from opentelemetry.util.types import Attributes
+from opentelemetry._logs import get_logger_provider
+from ldobserve._otel.configuration import _OTELConfiguration
+from ldobserve._util.dict import flatten_dict
+
+from opentelemetry.metrics import (
+    _Gauge as APIGauge,
+    Histogram as APIHistogram,
+    Counter as APICounter,
+    UpDownCounter as APIUpDownCounter,
+)
+
+_name = "launchdarkly-observability"
+
+
+class _ObserveInstance:
+    _project_id: str
+    _tracer: Tracer
+
+    _provider = get_logger_provider()
+    _logger = logging.getLogger(__name__)
+    _otel_configuration: _OTELConfiguration
+
+    _gauges: dict[str, APIGauge] = dict()
+    _counters: dict[str, APICounter] = dict()
+    _histograms: dict[str, APIHistogram] = dict()
+    _up_down_counters: dict[str, APIUpDownCounter] = dict()
+
+    @property
+    def log_handler(self) -> logging.Handler:
+        return self._otel_configuration.log_handler
+
+    def __init__(self, project_id: str, otel_configuration: _OTELConfiguration):
+        self._otel_configuration = otel_configuration
+
+        # Logger that will only log to OpenTelemetry.
+        self._logger.propagate = False
+        self._logger.addHandler(otel_configuration.log_handler)
+        self._project_id = project_id
+        self._tracer = trace.get_tracer(_name)
+
+    def record_exception(
+        self, error: Exception, attributes: typing.Optional[Attributes] = None
+    ):
+        span = trace.get_current_span()
+        if not span:
+            raise RuntimeError("H.record_exception called without a span context")
+
+        attrs = {}
+        if attributes:
+            addedAttributes = flatten_dict(attributes, sep=".")
+            attrs.update(addedAttributes)
+
+        span.record_exception(error, attrs)
```

Comment:
confirming that we're not providing a start_span / contextmanager API - would we instruct customers to use the opentelemetry package directly?

Reply:
I can add it.